kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/GwtNetworkServiceImpl.java

```diff
@@ -99,6 +114,76 @@ public static void updateNetInterfaceConfigurations(GwtNetInterfaceConfig config
         }
     }
 
+    public static List<GwtFirewallOpenPortEntry> findDeviceFirewallOpenPorts() throws GwtKuraException {
+        FirewallConfigurationService fcs = ServiceLocator.getInstance().getService(FirewallConfigurationService.class);
+        List<GwtFirewallOpenPortEntry> gwtOpenPortEntries = new ArrayList<>();
+
+        try {
+            List<NetConfig> firewallConfigs = fcs.getFirewallConfiguration().getConfigs();
+            if (firewallConfigs != null && !firewallConfigs.isEmpty()) {
+                for (NetConfig netConfig : firewallConfigs) {
+                    if (netConfig instanceof FirewallOpenPortConfigIP4) {
+                        FirewallOpenPortConfigIP4 firewallOpenPortConfigIP4 = (FirewallOpenPortConfigIP4) netConfig;
+                        logger.debug("findDeviceFirewallOpenPorts() :: adding new Open Port Entry: {}",
+                                firewallOpenPortConfigIP4.getPort());
+                        gwtOpenPortEntries.add(convertToGwtFirewallOpenPortEntry(firewallOpenPortConfigIP4));
+                    }
+                }
+            }
```

Comment:
I feel like this is the perfect spot to add a couple of early-returns to make the code more readable.

```suggestion
            if (firewallConfigs == null || firewallConfigs.isEmpty()) {
                return new ArrayList<>();
            }
            for (NetConfig netConfig : firewallConfigs) {
                if (!netConfig instanceof FirewallOpenPortConfigIP4) {
                    continue;
                }
                FirewallOpenPortConfigIP4 firewallOpenPortConfigIP4 = (FirewallOpenPortConfigIP4) netConfig;
                logger.debug("findDeviceFirewallOpenPorts() :: adding new Open Port Entry: {}",
                        firewallOpenPortConfigIP4.getPort());
                gwtOpenPortEntries.add(convertToGwtFirewallOpenPortEntry(firewallOpenPortConfigIP4));
            }
```

What do you think?